base64_json.go

```diff
@@ -23,7 +23,7 @@ func jsonSort(a *orderedmap.Pair, b *orderedmap.Pair) bool {
 	return jsonOrder[a.Key()] < jsonOrder[b.Key()]
 }
 
-func Encode(v interface{}) (*string, error) {
+func Base64JsonEncode(v interface{}) (*string, error) {
```

Comment:
I think this should not be public. If you can give me proof this is useful to the end user (it should not), this should be documented.

Reply:
Agreed, these methods should be private (and not documented).

This is the reason we had several packages internally, the result being that the `identity` package only contained the public methods that we _really_ want to expose.

Now that you merged all methods in the single `identity` package, we could use a strategy where internal methods (like this one) are tested with "internal" tests (defined in the same package), whereas public methods are tested in a separate `identity_test` package. See: https://medium.com/@matryer/5-simple-tips-and-tricks-for-writing-unit-tests-in-golang-619653f90742 for reference.
 
I would also like to point out that the current situation is confusing since our `*_test.go` files are sometimes using the `identity_test` package, and sometimes the `identity` package. We need to clarify that (maybe by applying the pattern describe in the blog post above).

Reply:
Interesting article! I have to admit I'm more of a fan of "package scope one-way tests", having different testing semantics looks a little off for me, but I can accomodate if need be. As mentioned in my comment above, it would also be possible (and IMO a clean way to Go) to put internally-only used packages in a `internal/` repo, where we can define specific tests, and export things that won't be usable anywhere else than this package... Maybe a BR is needed here? WDYT?

Reply:
Note that the use of an `internal/` package would do the trick too.